### PR TITLE
Add KONSOLE_VERSION to the list of preserved env vars

### LINKED ISSUE #1449: Curly underline ("undercurl") not working in Neovim + Konsole
**Describe the bug**
Curly underlines do not work inside of neovim, instead straight underlines are displayed.

**Steps how to reproduce the behaviour**
1. Open up **Konsole** from the host system
2. Enter a Fedora 39 container using toolbox
3. Set up a neovim config with LSP
4. Open a file in neovim with errors
5. Inspect the style of the underline

**Expected behaviour**
Curly underlines are displayed (this is what it looks like inside of distrobox).
image

**Actual behaviour**
Straight underline are displayed (this is what it looks like inside of toolbox).
image

**Screenshots**
See above.

**Output of `toolbox --version` (v0.0.90+)**
`toolbox version 0.0.99.5`

**Toolbx package info (`rpm -q toolbox`)**
`toolbox-0.0.99.5-2.fc39.x86_64`

**Output of `podman version`**
```
Client:       Podman Engine
Version:      4.9.0
API Version:  4.9.0
Go Version:   go1.21.6
Built:        Wed Jan 24 11:07:27 2024
OS/Arch:      linux/amd64
```

**Podman package info (`rpm -q podman`)**
`podman-4.9.0-1.fc39.x86_64`

**Info about your OS**
Fedora KDE 39

**Additional context**
I believe that this issue is specific to toolbox. I have tested this both inside of toolbox and distrobox and 
for some reason it works inside of distrobox but not inside of toolbox. When I run `printenv | grep TERM`, the 
output for both is:
```
COLORTERM=truecolor
TERM=xterm-256color
```
However when I run `export TERM="alacritty"` inside of toolbox, I have curly underlines. But setting `TERM` to `alacritty`
just after entering the container feels very _hacky_, therefore a proper solution would be much appreciated.

### PR DESCRIPTION
**Pull request by notfirefox**
As discussed [here](https://github.com/containers/toolbox/issues/1449#issuecomment-1951842319).

- Adds `KONSOLE_VERSION` to the list of preserved environment variables

Closes #1449

### CODE REVIEW

**Review by debarshiray — CHANGES_REQUESTED** [review]
Thanks for figuring this out, @notfirefox !  Could you please correct your Git authorship information with your full name and a real email address, unless you are worried about privacy?

File: src/pkg/utils/utils.go [hunk 1]
```diff
@@ -90,6 +90,7 @@ var (
 		"HISTIGNORE",
 		"HISTSIZE",
 		"HISTTIMEFORMAT",
+		"KONSOLE_VERSION",
```
Comment (debarshiray) [inline src/pkg/utils/utils.go:93]: It will be good to add some tests for `KONSOLE_VERSION` to `test/system/220-environment-variables.bats`.

**Comment by notfirefox** [comment]
> Thanks for figuring this out, @notfirefox ! Could you please correct your Git authorship information with your full name and a real email address, unless you are worried about privacy?

I'd prefer to keep the authorship information as it is because of privacy concerns.

File: src/pkg/utils/utils.go (hunk 1 above)
Comment (notfirefox) [reply src/pkg/utils/utils.go:93]: I have added some tests. I had some difficulties running them locally, but the bot seems to be happy.

image
[system-test-fedora-39/console](https://softwarefactory-project.io/zuul/t/local/build/334f5a2571684213a9bc6e5f7ddf93d7/console)

**Comment by debarshiray** [comment]
> > Thanks for figuring this out, @notfirefox ! Could you please correct your Git authorship information with your full name and a real email address, unless you are worried about privacy?
> 
> I'd prefer to keep the authorship information as it is because of privacy concerns.

I see.  A few things:

The `Signed-off-by` trailer has no meaning without a real full name and email address.

I could have made an exception for a one line change, like adding an environment variable, because it would be legally insignificant.  However, that's no longer the case with this pull request after the addition of the tests.  :)

We need a real full name and email address for tracking copyright holders in case a licensing issue crops up.  Note how projects as diverse as [GCC](https://gcc.gnu.org/dco.html), [GnuPG](https://www.gnupg.org/faq/HACKING.html), [Linux](https://www.kernel.org/doc/html/latest/process/1.Intro.html) and [Podman](https://github.com/containers/podman/blob/main/CONTRIBUTING.md) don't allow anonymous or pseudonymous contributions.

**Comment by notfirefox** [comment]
@debarshiray DCO related issues should be resolved now.

**Review by debarshiray — COMMENTED** [review]
Thanks for updating the pull request, @notfirefox !  Let me rebase it against `main`.

File: test/system/220-environment-variables.bats [hunk 2]
```diff
@@ -565,3 +565,190 @@ teardown() {
...
+
+@test "environment variables: KONSOLE_VERSION inside the default container" {
+  create_default_container
+
+  if [ "$KONSOLE_VERSION" = "" ]; then
+    # shellcheck disable=SC2030
+    KONSOLE_VERSION=230804
+  fi
+
+  export KONSOLE_VERSION
```
[the remark this answers is not in the document]
Comment (debarshiray) [reply test/system/220-environment-variables.bats]: I took the liberty to adjust this.

**Comment by debarshiray** [comment]
Thanks for working on this, @notfirefox ; and for reconsidering the privacy issues!  Much appreciated.

**Pull request merged**